File: Python/Ram.py
from Komponen import Komponen

class Ram(Komponen):
    def __init__(self, merk="", nama="", kapasitasGB=0, tipeDDR=""):
        super().__init__(merk, nama)
        self.kapasitasGB = kapasitasGB
        self.tipeDDR = tipeDDR

    def get_kapasitasGB(self):
        return self.kapasitasGB

    def get_tipeDDR(self):
        return self.tipeDDR

    def set_kapasitasGB(self, kapasitasGB):
        self.kapasitasGB = kapasitasGB

    def set_tipeDDR(self, tipeDDR):
        self.tipeDDR = tipeDDR

    def info(self):
        return f"RAM: {self.kapasitasGB}GB {self.tipeDDR} (Merk: {self.get_merk()})"

File: java/Ram.java
public class Ram extends Komponen {
    private int kapasitasGB;
    private String tipeDDR;

    public Ram(String merk, String nama, int kapasitasGB, String tipeDDR) {
        super(merk, nama);
        this.kapasitasGB = kapasitasGB;
        this.tipeDDR = tipeDDR;
    }

    public int getKapasitasGB() {
        return kapasitasGB;
    }

    public String getTipeDDR() {
        return tipeDDR;
    }

    public void setKapasitasGB(int kapasitasGB) {
        this.kapasitasGB = kapasitasGB;
    }

    public void setTipeDDR(String tipeDDR) {
        this.tipeDDR = tipeDDR;
    }

    public String info() {
        return "RAM: " + kapasitasGB + "GB " + tipeDDR + " (Merk: " + getMerk() + ")";
    }
}